### DISCUSSION TITLE
Generating outline command freezes with high CPU use

### DISCUSSION THREAD

**Discussion by davideocam** [winder/Universal-G-Code-Sender#2531] [Help]
I have installed UGS on a newer Windows 10 PC after my old laptop died (also W10). Everything works well, including Xbox controller for movement, running g-code for cutting etc: all seems good except for _generate outline_. Activating this results in a timer and very high CPU use. Command eventually times out, but CNC does not move at all to describe the outline path. This worked fine on my older setup. The only change is the PC. Any ideas? Thank you.

**Comment by breiler** [comment]
Can you post the gcode?
Which version of UGS are you using?

**Reply by davideocam** [reply]
Hi, thanks for your comment.
I'm using 2.1.7, and I tried 2.1.6, which exhibits the same issue.
I tried both these files and the problem is the same. They cut fine, it's only the outlining command that seems to be a problem.
[example g-code.zip](https://github.com/winder/Universal-G-Code-Sender/files/15474944/example.g-code.zip)
I have since installed UGS on another laptop (DELL Precision 3571 Windows 11) and have the same problem.
Thanks for your help.
.

**Reply by davideocam** [reply]
P.S. g-code generated from Easel. I have not tried from Fusion360, CREO etc.

**Reply by breiler** [reply]
Thanks, I managed to reproduce it. I am currently travelling so I don't know when I have the time to fix it. Thanks for taking your time reporting it.

**Reply by davideocam** [reply]
Many thanks for your update. I fully understand having other commitments! It's easy to work around temporarily with a skeleton/summary g-code to outline, (just to check the movement range and clearance before final machining). It might be a nice feature if the outlined path returned to zero (or equivalent starting position)?

**Reply by breiler** [reply]
I think that I have fixed this in the latest nightly build - available here:
https://github.com/winder/Universal-G-Code-Sender/releases/tag/nightly

**Reply by davideocam** [reply]
Good evening. I've just run a small profile cut for a plate and it worked fine on that. Thank you.
